falcon/media/urlencoded.py

```diff
@@ -25,13 +29,21 @@ class URLEncodedFormHandler(BaseHandler):
     """
 
     def __init__(self, keep_blank=True, csv=False):
-        self.keep_blank = keep_blank
-        self.csv = csv
+        self._keep_blank = keep_blank
+        self._csv = csv
+
+        # NOTE(kgriffs): To be safe, only enable the optimized protocol when
+        #   not subclassed.
+        if type(self) is URLEncodedFormHandler:
+            self._serialize_sync = self.serialize
+            self._deserialize_sync = self._deserialize
 
-    def serialize(self, media, content_type):
+    # NOTE(kgriffs): Make content_type a kwarg to support the
+    #   Request.render_body() shortcut optimization.
+    def serialize(self, media, content_type=None) -> bytes:
         # NOTE(vytas): Setting doseq to True to mirror the parse_query_string
         # behaviour.
-        return urlencode(media, doseq=True)
+        return urlencode(media, doseq=True).encode()
```

Comment:
During testing I discovered and fixed this bug.

Reply:
:man_facepalming: